weblate/trans/models/suggestion.py

```diff
@@ -44,15 +45,22 @@ def add(self, unit, target, request, vote=False):
         """Create new suggestion for this unit."""
         user = request.user
 
-        same = self.filter(
-            target=target,
-            content_hash=unit.content_hash,
-            language=unit.translation.language,
-            project=unit.translation.component.project,
-        )
+        try:
+            same = self.get(
+                target=target,
+                content_hash=unit.content_hash,
+                language=unit.translation.language,
+                project=unit.translation.component.project,
+            )
+
+            if same.user == user:
```

Comment:
```suggestion
            if same.user == user or not vote:
```

This is actually better fix for voting disabled case.